### ISSUE TITLE
Add hierarchal view to explorer

### ISSUE THREAD

**Issue by ScriptPup** [ScriptPup/sqltools-teradata-driver#9]
I received the below feature request via email from a user.

This is the View that I currently have in VS Code. However, I need a hierarchical overview, just like in the teradata studio. Also, I can see Databases, but not 

![image](https://user-images.githubusercontent.com/6163646/148595838-7e514722-d05f-4686-9f2d-7c2d9d974e1b.png)

**Additional context:**
Databases and users in Teradata can form a hierarchy, i.e. a db/user can be created within another db/user. See the SQL example below. 

```
CREATE DATABASE parent
AS PERMANENT = 60e8,
    SPOOL = 120e8;
    
CREATE DATABASE child FROM parent
AS PERMANENT = 60e7, -- notice how space gets reduced as it needs to fit into the parent db space
    SPOOL = 120e7;

CREATE DATABASE grandchild FROM child
AS PERMANENT = 60e6,
    SPOOL = 120e6;   

-- and now, let’s display the hierarchy

WITH RECURSIVE JER (LEVEL, DATABASENAME, OWNERNAME, ROUTE)
AS
(
SELECT 1 LEVEL,
       DATABASENAME,
       OWNERNAME,
       CAST('/'||TRIM(DATABASENAME) AS VARCHAR(128)) ROUTE
  FROM DBC.DATABASES
 WHERE DATABASENAME='DBC'
UNION ALL
SELECT JER.LEVEL + 1,
       D.DATABASENAME,
       D.OWNERNAME,
       JER.ROUTE || '/' || TRIM(D.DATABASENAME) 
  FROM DBC.DATABASES D,
       JER
 WHERE d.DATABASENAME<>'DBC'
   AND d.OWNERNAME = JER.DATABASENAME
)
SELECT LEVEL, 
       DATABASENAME, 
       OWNERNAME, 
       ROUTE
  FROM JER
 ORDER BY 4
```

**Comment by jimboug** [comment]
I support this enhancement as well

**Comment by ScriptPup** [comment]
Implemented - Changes will be included with V0.4.0 build.

<img width="214" alt="image" src="https://user-images.githubusercontent.com/6163646/233803542-b87f1a72-e007-4d0c-b0ad-f70be3ef489f.png">

Currently, there's no `Users` object type supported by the parent project sql-tools. 
Due to this, this **might** make it impossible to view databases nested under users, if that's possible? 

Currently the query used to pull the ROOT level databases is:
```
SELECT 
    DatabaseName AS "label"
    ,DatabaseName as "name"
    ,DatabaseName AS "database"
    ,'${ContextValue.DATABASE}' AS "type"
    ,'database' AS "detail"
    ,'database' AS "iconId"
  FROM DBC.DBase
  WHERE 
    RowType = 'D'
    AND OwnerName = 'DBC' --"Root" level databases only, don't pull nested DBs here
  ;
```
As far as I've found, 'D' is the correct type for Database... 

I'm going to be working on adding a toggle for the connection to display users as databases for #16. I think that should mitigate this possible issue... I think that actually will also solve for #11 as well.
